test/lib/commands/install.js

```diff
@@ -400,3 +408,148 @@ t.test('should show install keeps dirty --workspace flag', async t => {
   assert.packageDirty('node_modules/abbrev@1.1.0')
   assert.packageInstalled('node_modules/lodash@1.1.1')
 })
+
+t.test('devEngines', async t => {
+  const globals = {
+    'process.platform': 'linux',
+    'process.arch': 'x86',
+    'process.version': 'v1337.0.0',
+  }
+
+  t.test('should utilize devEngines success case', async t => {
+    const { npm, joinedFullOutput } = await loadMockNpm(t, {
+      globals,
+      prefixDir: {
+        'package.json': JSON.stringify({
+          name: 'test-package',
+          version: '1.0.0',
+          devEngines: {
+            runtime: {
+              name: 'node',
+            },
+          },
+        }),
+      },
+    })
+    await npm.exec('install', [])
+    t.matchSnapshot(joinedFullOutput())
+  })
+
+  t.test('should utilize devEngines failure case', async t => {
+    const { npm, joinedFullOutput } = await loadMockNpm(t, {
+      globals,
+      prefixDir: {
+        'package.json': JSON.stringify({
+          name: 'test-package',
+          version: '1.0.0',
+          devEngines: {
+            runtime: {
+              name: 'nondescript',
+            },
+          },
```

Comment:
It might be good to test the runtime or package manager's version being out of range as well, unless I overlooked more exhaustive tests elsewhere.

```javascript
          devEngines: {
            runtime: {
              name: 'node',
              version: '>=2048.0.0'
            },
          },
```

Reply:
@hashtagchris Hey! Yep there are more exhaustive tests in the accompanying PR found here: https://github.com/npm/npm-install-checks/blob/83e7c96eeda57d612edbdb0f21863aa6fb7a7ace/test/check-dev-engines.js